2016/05/what-is-react.js-and-why-use-react.md

```diff
@@ -0,0 +1,57 @@
+---
+title: What is React.js and why we use it?
+tags: Frontend, React.js, Redux
+---
+
+We are using [React.js](https://facebook.github.io/react/index.html), a JavaScrip library for creating user interfaces, we use [React.js](https://facebook.github.io/react/index.html) because is simple, clean and elegant code, and easy to maintain. For official documentation look at [Getting Started](https://facebook.github.io/react/docs/getting-started.html) or [Tutorial](https://facebook.github.io/react/docs/tutorial.html) pages.
+
+**From my point of view, React has the following advantages:**
+
+* Good if you what to build large applications with data that changes over time.
+* **Virtual DOM**. React abstracts away the DOM (a programming API for HTML and XML documents), you diff that with the actual DOM, and only update what changed. This makes updates faster.
+* Perfect to render UI components.
+* Offers incredible performance and server side rendering.
+* You can always change how the component will render modifying a source file.
+* Building Javascript and HTML into JSX makes components easily understandable.
+* React.js is declarative, which means that React conceptually hits the “refresh” button any time data changes.
+* You can use React.js seamlessly with Node.js or AngularJS.
+* Uni-directional data flow.
+* The ability of return an html component from a function.
+* The child notify its parent of a state change with a callback pass as one of its properties.
```

Comment:
This is not quite clear to me... idk.. maybe:
The child notify its parent when a change occurs with a callback passed as one of its properties. 
Is the above sentence correct? @iLeonelPerea